conn.go

```diff
@@ -38,11 +39,15 @@ func Dial(config *netlink.Config) (*Conn, error) {
 		return nil, err
 	}
 
-	return newConn(c), nil
+	return NewConn(c), nil
 }
 
-// newConn is the internal constructor for Conn, used in tests.
-func newConn(c conn) *Conn {
+// NewConn creates a Conn that wraps an existing *netlink.Conn for
+// generic netlink communications.
+//
+// NewConn is primarily useful for tests. Most applications should use
+// Dial instead.
+func NewConn(c conn) *Conn {
```

Comment:
Avoid exporting this if you can, or if you can't, you'll want to pass some exported type here.  The reason I do it is for use in `genltest`: https://github.com/mdlayher/genetlink/blob/master/genltest/genltest.go#L45.